### PR TITLE
Eus bug fix

### LINKED ISSUE #1578: enduser_setup can fail to cleanup tcp listen sockets
### Expected behavior
enduser_setup.start() doesn't always succeed in being called again.  Calling stop() should cleanup properly, and you should be able to call start() again and have it "work"

### Actual behavior
Sometimes, after successfully connecting to the chosen AP, if you then interactively try and start again, you can leak the tcp socket, and are unable to restart the process without resetting.

### Test code
```Lua
function es_ok()
        print(string.format("Connected to wifi ssid: %s our IP: %s", wifi.sta.getconfig(), wifi.sta.getip()))
end
function es_bad(err, str)
        print("es failed, err: ", err, " str: ", str)
end
function es_dbg(str)
        print("es dbg: ", str)
end
> wifi.setmode(wifi.SOFTAP)
> enduser_setup.start(es_ok, es_bad, es_dbg)
es dbg: 	340: 	enduser_setup_check_station_start
es dbg: 	1219: 	enduser_setup_ap_start
es dbg: 	1348: 	enduser_setup_dns_start
es dbg: 	1154: 	enduser_setup_http_start
es dbg: 	1163: 	http_start bind failed <<<
es dbg: 	1526: 	enduser_setup_stop
es dbg: 	1211: 	enduser_setup_station_stop
es dbg: 	1400: 	enduser_setup_dns_stop
es dbg: 	1200: 	enduser_setup_http_stop
es dbg: 	1317: 	enduser_setup_free
> enduser_setup.stop()
> enduser_setup.stop()
> enduser_setup.stop()
> wifi.setmode(wifi.SOFTAP)
> enduser_setup.start(es_ok, es_bad, es_dbg)
es dbg: 	340: 	enduser_setup_check_station_start
es dbg: 	1219: 	enduser_setup_ap_start
es dbg: 	1348: 	enduser_setup_dns_start
es dbg: 	1154: 	enduser_setup_http_start
es dbg: 	1163: 	http_start bind failed   <<<
es dbg: 	1526: 	enduser_setup_stop
es dbg: 	1211: 	enduser_setup_station_stop
es dbg: 	1400: 	enduser_setup_dns_stop
es dbg: 	1200: 	enduser_setup_http_stop
es dbg: 	1317: 	enduser_setup_free
> enduser_setup.stop()
> 
```
### NodeMCU version
Which branch are you on? If you know the Git revision then add it here as well.
1.5.4.1-master_20161001  + local changes to enable ENDUSER_SETUP_DEBUG_ENABLE

### Hardware
Describe which ESP8266 device you use and document any special hardware setup 
required to reproduce the problem.
the "amica" board, esp12e+cp2104 module.

### PR DESCRIPTION
**Pull request by jfollas**
Fixes #1578.

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/en/*`.

FIX: Ensure that station is enabled before doing network scan (this seems to be the cause of #1578)
FIX: Check if state == NULL on enduser_setup.stop()
IMPROVEMENT: Look for both enduser_setup.html.gz and enduser_setup.html as valid filenames, for semantic reasons
IMPROVEMENT: Show a generic status message on HTML page if failure occurred before user even submitted data (i.e., would have come from previously stored config when the STA was started by EUS)

### CODE REVIEW

**Comment by marcelstoer** [comment]
Good stuff Jason, thanks!

Even though it fixes bugs I suggest we wait with merging until after the [upcoming master drop is done](https://github.com/nodemcu/nodemcu-firmware#releases) as it's much more than a simple bug fix.

File: docs/en/modules/enduser-setup.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 # enduser setup Module
 | Since  | Origin / Contributor  | Maintainer  | Source  |
 | :----- | :-------------------- | :---------- | :------ |
-| 2015-09-02 | [Robert Foss](https://github.com/robertfoss) | [Robert Foss](https://github.com/robertfoss) | [enduser_setup.c](../../../app/modules/enduser_setup.c)|
+| 2015-09-02 | [Robert Foss](https://github.com/robertfoss), [Jason Follas](https://github.com/jfollas) | [Robert Foss](https://github.com/robertfoss) | [enduser_setup.c](../../../app/modules/enduser_setup.c)|
```
Comment (marcelstoer) [inline docs/en/modules/enduser-setup.md]: The use of the terms "Origin / Contributor" may not be very obvious here, I agree. It isn't documented anywhere either. However, we've been using it like this:
- origin: where the code originally branched off from. Many of our modules where ported to NodeMCU from somewhere, often from Arduino.
- contributor: the developer(s) who contributed the original NodeMCU code (i.e. often ported code).

→ "Origin / Contributor" are the same for NodeMCU modules like EUS which started from scratch
→ "Origin / Contributor" usually don't change - even though you greatly contributed to EUS, thanks again 🍺 

Maybe we should change this pattern.

Even better would be if you allowed us to list you as the maintainer 😃 It seems @robertfoss didn't have enough spare time recently to maintain it.

Comment (jfollas) [inline docs/en/modules/enduser-setup.md]: I wasn't trying to take credit or ownership away from @robertfoss , but just add accountability. I'm fine with not having anything modified in the docs.

Comment (marcelstoer) [inline docs/en/modules/enduser-setup.md]: I know you didn't! It's good that you added your name to the source code in an earlier contribution.
If you have good ideas as for how to give credit to those who contribute significantly I'm sure we'd welcome those as well. [GitHub's charts](https://github.com/nodemcu/nodemcu-firmware/graphs/contributors) are nice but just a machine's point of view of course.

**Pull request merged**